vars/copyContainer.groovy

```diff
@@ -10,34 +10,66 @@
+    destination_image_no_tag = destination_image.split(':')[0]
+    destination_registry = args.destinationRegistry
 
     if (args.destinationRegistry == 'opensearchstaging' || args.destinationRegistry == 'opensearchproject') {
         def dockerJenkinsCredential = args.destinationRegistry == 'opensearchproject' ? "jenkins-production-dockerhub-credential" : "jenkins-staging-dockerhub-credential"
         withCredentials([usernamePassword(credentialsId: dockerJenkinsCredential, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
-            def dockerLogin = sh(returnStdout: true, script: "set +x && echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin").trim()
-            sh """gcrane cp ${args.sourceRegistry}/${args.sourceImage} ${args.destinationRegistry}/${args.destinationImage}; docker logout"""
+            sh("set +x && echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin")
+            craneCopy()
         }
     }
+
     if (args.destinationRegistry == 'public.ecr.aws/opensearchproject') {
         withCredentials([
             string(credentialsId: 'jenkins-artifact-promotion-role', variable: 'ARTIFACT_PROMOTION_ROLE_NAME'),
             string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT')])
             {
                 withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
-                    def ecrLogin = sh(returnStdout: true, script: "set +x && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${args.destinationRegistry}").trim()
-                    sh """gcrane cp ${args.sourceRegistry}/${args.sourceImage} ${args.destinationRegistry}/${args.destinationImage}; docker logout ${args.destinationRegistry}"""
+                    sh("set +x && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${args.destinationRegistry}")
+                    craneCopy()
                 }
             }
     }
+
     if(args.destinationRegistry == 'public.ecr.aws/opensearchstaging') {
-            def ecrLogin = sh(returnStdout: true, script: "set +x && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${args.destinationRegistry}").trim()
-            sh """gcrane cp ${args.sourceRegistry}/${args.sourceImage} ${args.destinationRegistry}/${args.destinationImage}; docker logout ${args.destinationRegistry}"""
+        sh("set +x && aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${args.destinationRegistry}")
+        craneCopy()
+    }
+
+}
+
+def craneCopy() {
+
+    if (all_tags == true) {
+        if (destination_registry.contains('opensearchstaging') && destination_image_no_tag.equals('ci-runner')) {
```

Comment:
Confused with image name and repository name:
https://hub.docker.com/r/opensearchstaging/ci-runner/tags
https://gallery.ecr.aws/opensearchstaging/ci-runner
When you say registry it also includes repo name right?

Reply:
yes, that is not ideal but it is by design following the initial design of the docker copy workflow.

Reply:
docker.io/opensearchstaging vs. public.ecr.aws/opensearchstaging, technically this is reg+repo name overall.

Reply:
Can we atleast make the variables name right then?
`destination_image_no_tag` -> `repository_name`?

Reply:
That one is correct var name.
destination image is: `ci-runner:centos-cli`
destination_image_no_tag is: `ci-runner`

You are mistaken the repo name as image name.

Reply:
public.ecr.aws/opensearchstaging/ci-runner:centos-cli
|---registry---|----repo name----|--image--|---tag---|